### ISSUE TITLE
dnsmasq not starting when /sbin/resolvconf is missing on CentOS 7 ARM

### ISSUE THREAD

**Issue by diggnsaegg** [pi-hole/pi-hole#3309]
**In raising this issue, I confirm the following:** `{please fill the checkboxes, e.g: [X]}`

- [X] I have read and understood the [contributors guide](https://github.com/pi-hole/pi-hole/blob/master/CONTRIBUTING.md).
- [X] The issue I am reporting can be *replicated*.
- []X The issue I am reporting isn't a duplicate (see [FAQs](https://github.com/pi-hole/pi-hole/wiki/FAQs), [closed issues](https://github.com/pi-hole/pi-hole/issues?utf8=%E2%9C%93&q=is%3Aissue%20is%3Aclosed%20), and [open issues](https://github.com/pi-hole/pi-hole/issues)).

**How familiar are you with the the source code relevant to this issue?:**

`5`

---
**Expected behaviour:**

`Pihole DNS service should run`

**Actual behaviour:**

`Pihole DNS service is not running because /sbin/resolvconf is missing`

**Steps to reproduce:**

`{Detailed steps of how we can reproduce this}`

**Debug token provided by [uploading `pihole -d` log](https://discourse.pi-hole.net/t/the-pihole-command-with-examples/738#debug):**

`not neccessary`

**Troubleshooting undertaken, and/or other relevant information:**

I run pihole on centos7.7 ARM (Rpi 3). I added an manual entry to my /etc/hosts file and executed "pihole restartdns" - this leads into nothing, it just waits, dunno I there is a timeout, I sent a sigterm and started investigating.

` systemctl status pihole-FTL.service
● pihole-FTL.service - LSB: pihole-FTL daemon
   Loaded: loaded (/etc/rc.d/init.d/pihole-FTL; bad; vendor preset: disabled)
   Active: active (exited) since Thu 2020-04-30 17:38:01 CEST; 26min ago
     Docs: man:systemd-sysv-generator(8)

Jan 01 01:00:23 pihole.local systemd[1]: Starting LSB: pihole-FTL daemon...
Jan 01 01:00:23 pihole.local pihole-FTL[653]: Not running
Jan 01 01:00:23 pihole.local pihole-FTL[653]: /etc/rc.d/init.d/pihole-FTL: line 51: /sbin/resolvconf: No such file or directory
Jan 01 01:00:23 pihole.local su[707]: (to pihole) root on none
Apr 30 17:38:01 pihole.local pihole-FTL[653]: FTL started!
Apr 30 17:38:01 pihole.local systemd[1]: Started LSB: pihole-FTL daemon.
`

As you can see the resolvconf binary is not found on my system and not even installable. I don't know if Red Hat ditched it recently because I never had any problems like this.
My solution to this was simply commenting out line 51 in /etc/rc.d/init.d/pihole-FTL
 Works like a charm but I thought it is maybe worth a bug report. If not just close it.

Regards

Dan

**Comment by dschaper** [comment]
Thanks for the report. Version 5 will not require `resolvconf`, we wont make any changes to `/etc/resolv.conf`. So hang in there a little bit longer and this should be resolved.

**Comment by bcambl** [comment]
CentOS 7 is only supported on x86_64 due to some missing dependencies in the past (EPEL). 
see: https://docs.pi-hole.net/main/prerequesites/

@diggnsaegg did you have any issues with the install / did you need to do any additional manual steps? Considering you were able to troubleshoot the issue, I assume you are capable 👍 

`/sbin/resolvconf` is not available on CentOS. `/usr/sbin/resolvconf` is provided via `systemd` on Fedora, either way.. I am surprised this has not been caught earlier as the commit is 2yrs old!

**Comment by bcambl** [comment]
```
[bcambl@localhost ~]$ stat /sbin/resolvconf
  File: /sbin/resolvconf -> ../bin/resolvectl
  Size: 17              Blocks: 0          IO Block: 4096   symbolic link
Device: b304h/45828d    Inode: 6435731     Links: 1
Access: (0777/lrwxrwxrwx)  Uid: (    0/    root)   Gid: (    0/    root)
Context: system_u:object_r:bin_t:s0
Access: 2020-03-26 09:24:10.000000000 -0600
Modify: 2020-03-26 09:24:10.000000000 -0600
Change: 2020-02-06 12:35:26.548804356 -0600
 Birth: 2020-02-06 12:35:26.546804356 -0600
[bcambl@localhost ~]$ cat /etc/redhat-release
Fedora release 31 (Thirty One)
[bcambl@localhost ~]$
```

**Comment by diggnsaegg** [comment]
> @diggnsaegg did you have any issues with the install / did you need to do any additional manual steps? Considering you were able to troubleshoot the issue, I assume you are capable 👍

Thanks, no this install is running for quite a time lol, but I updated the system recently maybe this was the point, where CentOS ditched resolvconf, I dont know.


`tune2fs -l /dev/mmcblk0p3 | grep 'Filesystem created:'
Filesystem created:       Tue May  1 13:30:44 2018
`

Tomorrow it it's 2nd birthday :D

As I said, never had any problems, just needed to add Remi php72 repo.

**Comment by diggnsaegg** [comment]
Even in CentOS 8.1 (x86_x64) release resolvconf is still available lol

` stat /sbin/resolvconf
  File: /sbin/resolvconf -> ../bin/resolvectl
  Size: 17              Blocks: 0          IO Block: 4096   symbolic link
Device: fd00h/64768d    Inode: 532487      Links: 1
Access: (0777/lrwxrwxrwx)  Uid: (    0/    root)   Gid: (    0/    root)
Context: system_u:object_r:bin_t:s0
Access: 2020-04-30 21:35:30.307836955 +0200
Modify: 2020-04-09 23:52:20.000000000 +0200
Change: 2020-04-22 21:11:53.325113218 +0200
 Birth: -
 dan@k8s  ~  cat /etc/redhat-release
CentOS Linux release 8.1.1911 (Core)
`

**Comment by bcambl** [comment]
Yes, CentOS8 is MUCH closer to Fedora than CentOS7 so it should work better once the changes it requires are merged with v5.0.

**Comment by diggnsaegg** [comment]
Thanks for the information. I already plan to migrate to Fedora or run  pihole with podman as a container.

**Comment by dschaper** [comment]
Fixed in #3207

**Comment by DL6ER** [comment]
Pi-hole v5.0 has just been released.
